package/com.unity.formats.usd/Tests/Editor/USDPayloadComponentTests.cs

```diff
@@ -218,14 +218,20 @@ public IEnumerator ChangingUsdPayloadStateFromUnloadedToLoaded_MarksSceneDirty()
             Assert.IsFalse(m_UnityScene.isDirty, "Scene should not be dirty after after saving");
 
             // Change payload state and wait one frame to ensure Update runs
-            m_usdRoot.GetComponentInChildren<UsdPayload>().Load();
-            yield return null;
+            var payloadRoot = m_usdRoot.GetComponentInChildren<UsdPayload>();
+            var payloadRootChildCount = payloadRoot.transform.childCount;
+
+            payloadRoot.Load();
+            yield return new WaitUntil(() => m_UnityScene.isDirty == true);
```

Comment:
The original test was to address a bug where the scene was not set to dirty at all after changing the payload setting. In that case, this modification to the test would run forever (or presumably until a UTR timeout, which would mean no useful information from the failure). 

However it looks like we can't rely on UsdPayload APIs to correctly report their own status either (`UsdPayload.IsLoaded()` will return true _before_ it's actually been loaded, seems weird to me and should be reviewed) so we should find a more stable way to wait on whether the payload change has been registered before we check whether the scene is dirty. (and also probably change the test name back to `_MarksSceneDirty`, because that's the behaviour the test was written for)

Reply:
I decided to keep the `Generate` / `Destroys` tests and bring back the 2 dirties tests as I agree with your point as well,

But I didnt really a "clean" way to test this, so I decided to change it to a time out, and make sure that the scene is marked within 3 seconds - I wish Unity's `WaitUntil` had a way add a timeout value, but sadly, that doesnt seem to be the case - let me know what you think about this method

Reply:
I think let's wait on a number of frames instead of a time- 3 seconds is a long time for a test to run even if it is failing, and a shorter test time risks becoming unstable in the future if Unity execution gets slower. The expected behaviour is that the payload should be loaded in the next frame, so waiting 3 or 4 frames should be enough to confirm this, and if not then we have bigger issues :D

Reply:
I think we should add the timeout/ breakout in these other two tests too- in the (hopefully unlikely) scenario that this breaks in future and the tests are run in a different order, these will get stuck in the loop of waiting for the bool to change and not diagnose the actual issue which the other tests would flag.